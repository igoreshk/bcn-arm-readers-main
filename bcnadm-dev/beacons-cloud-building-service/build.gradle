plugins {
    id 'beacons.common-service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-starter-oauth2')
    implementation('org.springframework.amqp:spring-rabbit')
    implementation('org.apache.commons:commons-math3:3.6.1')
    implementation('org.geotools:gt-main:26.1')
}

jacocoTestCoverageVerification {
    violationRules {
        rule {
            limit {
                minimum = 0.80
            }
        }
    }
}
