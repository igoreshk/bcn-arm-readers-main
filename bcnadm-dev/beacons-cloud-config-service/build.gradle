plugins {
    id 'beacons.service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-config-server')
    implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-netflix-eureka-client'
}
