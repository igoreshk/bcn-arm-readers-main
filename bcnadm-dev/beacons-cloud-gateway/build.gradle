plugins {
    id 'beacons.service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-starter-config')
    implementation('org.springframework.cloud:spring-cloud-starter-netflix-eureka-client')
    implementation('org.springframework.cloud:spring-cloud-starter-openfeign')
    implementation('org.springframework.boot:spring-boot-starter-web') {
        exclude group: 'org.springframework.boot', module: 'spring-boot-starter-tomcat'
    }

    implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-oauth2'
    implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-netflix-zuul'
    implementation('org.springframework.cloud:spring-cloud-starter-security')

    compile "javax.servlet:javax.servlet-api:4.0.0"

    testImplementation('org.springframework.boot:spring-boot-starter-test')
    testImplementation('org.springframework.security:spring-security-test')
}

jacocoTestCoverageVerification {
    violationRules {
        rule {
            limit {
                minimum = 0.09
            }
        }
    }
}
