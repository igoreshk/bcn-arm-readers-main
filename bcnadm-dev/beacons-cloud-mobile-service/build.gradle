plugins {
    id 'beacons.common-service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-starter-oauth2')
    implementation('org.springframework.kafka:spring-kafka:2.5.1.RELEASE')
}

jacocoTestCoverageVerification {
    violationRules {
        rule {
            limit {
                minimum = 0.5
            }
        }
    }
}
