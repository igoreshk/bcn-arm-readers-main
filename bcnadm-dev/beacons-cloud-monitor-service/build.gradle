plugins {
    id 'beacons.common-service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-starter-oauth2')
    implementation('org.apache.commons:commons-math3:3.6.1')

    implementation('org.springframework.kafka:spring-kafka:2.5.1.RELEASE')
    implementation('org.apache.curator:curator-client:5.0.0') {
        exclude group: 'org.apache.zookeeper', module: 'zookeeper'
    }
    implementation('org.apache.curator:curator-x-async:5.0.0') {
        exclude group: 'org.apache.curator', module: 'curator-framework'
        exclude group: 'org.apache.zookeeper', module: 'zookeeper'
    }
    implementation group: 'org.apache.curator', name: 'curator-framework', version: "5.0.0"
    implementation group: 'org.apache.zookeeper', name: 'zookeeper', version: '3.6.1'
    implementation group: 'com.google.code.gson', name: 'gson', version: '2.8.6'

    implementation(group: 'io.dropwizard.metrics', name: 'metrics-core', version: '4.1.9')
    implementation(group: 'io.dropwizard.metrics', name: 'metrics-graphite', version: '4.1.9')

    implementation('org.geotools:gt-main:26.1')

    testImplementation('org.springframework.kafka:spring-kafka-test:2.5.1.RELEASE')
    testImplementation group: 'org.assertj', name: 'assertj-core', version: '3.16.1'
}

jacocoTestCoverageVerification {
    violationRules {
        rule {
            limit {
                minimum = 0.74
            }
        }
    }
}
