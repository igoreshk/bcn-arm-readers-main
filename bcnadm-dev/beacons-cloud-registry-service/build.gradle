plugins {
    id 'beacons.service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-starter-config')
    implementation('org.springframework.cloud:spring-cloud-starter-eureka-server:1.4.7.RELEASE') {
        exclude group: 'org.springframework.boot', module: 'spring-boot-starter-tomcat'
    }
}
