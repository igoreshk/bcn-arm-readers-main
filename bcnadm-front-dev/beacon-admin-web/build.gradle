plugins {
    id "com.github.node-gradle.node" version "2.2.4"
}

node {
    version = '12.13.0'
    download = true
    npmVersion = '6.12.0'
}

task installDependencies(type: NpmTask) {
    args = ['install']
}

task checkStyleSheets(type: NpmTask) {
    args = ['run', 'stylelint']
}
checkStyleSheets.dependsOn(installDependencies)

task checkMain(type: NpmTask) {
    args = ['run', 'lint-main']
}
checkMain.dependsOn(checkStyleSheets)

task checkTest(type: NpmTask) {
    args = ['run', 'lint-test']
}
checkTest.dependsOn(checkMain)

task test(type: NpmTask) {
    args = ['test']
}
test.dependsOn(checkTest)

// build whole frontend and
task build(type: NpmTask) {
    args = ['run', 'build']
}
build.dependsOn(test)

task start(type: NpmTask) {
    args = ['start']
}

task clean() {
    delete 'dist'
    delete 'build'
    delete 'out'
}
