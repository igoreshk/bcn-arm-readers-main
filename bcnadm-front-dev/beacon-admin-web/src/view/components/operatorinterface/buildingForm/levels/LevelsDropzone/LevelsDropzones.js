import { LevelImageService } from 'src/service/LevelImageService';
import React, { Component } from 'react';
import { Button } from '@material-ui/core';
import { Scrollbars } from 'react-custom-scrollbars';
import { Field } from 'redux-form';
import PropTypes from 'prop-types';

import { LEVEL_FIELD } from 'src/consts/FormsNames';
import LevelField from '../LevelField';

export default class LevelsDropzones extends Component {
  /**
   * After render levels (after adding new, for example) - scroll to last.
   */
  componentDidUpdate() {
    if (this.shouldScrollToBottom) {
      this.scrollToBottom();
      this.shouldScrollToBottom = false;
    }
  }
  lastLevelElement = React.createRef();
  shouldScrollToBottom = false;

  addNewLevelToModel = () => {
    const { levels, updateLevels } = this.props;
    const newLevelNumber = levels.length !== 0 && levels[levels.length - 1] ? levels[levels.length - 1].number + 1 : 1;
    levels.push({
      number: newLevelNumber,
      image: undefined
    });
    updateLevels(levels);
    this.shouldScrollToBottom = true;
  };

  changeLevelNumber = (levelIndex, newNumber) => {
    const { levels, updateLevels, changeStatusOfSubmit } = this.props;
    const unqieValue = this.checkValueForUnique(newNumber, 0);
    changeStatusOfSubmit(!unqieValue);
    const changedLevels = levels;
    changedLevels[levelIndex].number = Number(newNumber);
    updateLevels(changedLevels);
  };

  checkValueForUnique = (value, maxNumUniqueValue = 1) => {
    const { levels } = this.props;
    const numUniqueValue = levels.filter((level) => level.number === value).length;
    if (numUniqueValue > maxNumUniqueValue) {
      return false;
    }
    return true;
  };

  changeLevelImage = (levelIndex, newImage) => {
    const { levels, updateLevels } = this.props;
    const changedLevels = levels;
    changedLevels[levelIndex].image = newImage;
    updateLevels(changedLevels);
  };

  scrollToBottom = () => {
    const scrollToStartOfElement = true;
    this.lastLevelElement.current.scrollIntoView(scrollToStartOfElement);
  };

  renderLevelDropzones = () => {
    return this.props.levels.map((level, idx) => (
      <Field
        key={idx}
        component={LevelField}
        name={`${LEVEL_FIELD}[${idx}]`}
        params={{
          unique: this.checkValueForUnique(level.number),
          entityId: level.entityId,
          number: level.number,
          image: level.image,
          levelIndex: idx,
          imageLink: level.entityId ? LevelImageService.getLevelImageLink(level.entityId) : null
        }}
        onRemoveLevelClick={this.props.removeLevelField}
        changeLevelNumber={this.changeLevelNumber}
        changeLevelImage={this.changeLevelImage}
        translate={this.props.translate}
      />
    ));
  };

  render() {
    return (
      <div className="levelsDropzones">
        <Scrollbars className="scrollbar">
          <div className="scrollbarContent">
            {this.renderLevelDropzones()}
            <div ref={this.lastLevelElement} />
          </div>
        </Scrollbars>
        <div className="addLevelButtonContainer">
          <Button onClick={this.addNewLevelToModel} color="primary">
            {this.props.translate('building.addLevel')}
          </Button>
        </div>
      </div>
    );
  }
}

LevelsDropzones.propTypes = {
  levels: PropTypes.arrayOf(PropTypes.object),
  translate: PropTypes.func,
  updateLevels: PropTypes.func,
  removeLevelField: PropTypes.func,
  changeStatusOfSubmit: PropTypes.func
};
