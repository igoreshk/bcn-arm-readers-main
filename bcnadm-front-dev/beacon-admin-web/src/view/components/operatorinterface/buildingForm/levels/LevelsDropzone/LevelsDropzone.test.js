import React from 'react';
import { shallow } from 'enzyme/build';
import LevelsDropzones from './LevelsDropzones';

const props = {
  changeStatusOfSubmit: jest.fn(),
  translate: jest.fn(),
  updateLevels: jest.fn(),
  removeLevelField: jest.fn(),
  levels: [{ number: 1 }, { number: 2 }]
};

const wrapper = shallow(<LevelsDropzones {...props} />);

describe('Testing LevelsDropzone', () => {
  it('Check matching snapshot', () => expect(wrapper).toMatchSnapshot());

  it('checkValueForUnique will be call with correct arguments when call changeLevelNumber', () => {
    const checkValueForUnique = jest.spyOn(wrapper.instance(), 'checkValueForUnique');
    wrapper.instance().changeLevelNumber(1, 1);
    expect(checkValueForUnique).toHaveBeenCalledWith(1, 0);
  });

  it('changeStatusOfSubmit will be called when call changeLevelNumber', () => {
    const changeStatusOfSubmit = jest.spyOn(wrapper.instance().props, 'changeStatusOfSubmit');
    wrapper.instance().changeLevelNumber(1, 1);
    expect(changeStatusOfSubmit).toHaveBeenCalled();
  });

  it('checkValueForUnique will be called when rendering', () => {
    const checkValueForUnique = jest.spyOn(wrapper.instance(), 'checkValueForUnique');
    expect(checkValueForUnique).toHaveBeenCalled();
  });

  it('updateLevels will be called with correct argument', () => {
    const updateLevels = jest.spyOn(wrapper.instance().props, 'updateLevels');
    wrapper.instance().changeLevelNumber(1, 1);
    expect(updateLevels).toHaveBeenNthCalledWith(1, [{ number: 1 }, { number: 1 }]);
    expect(updateLevels).toHaveBeenNthCalledWith(2, [{ number: 1 }, { number: 1 }]);
  });
});
