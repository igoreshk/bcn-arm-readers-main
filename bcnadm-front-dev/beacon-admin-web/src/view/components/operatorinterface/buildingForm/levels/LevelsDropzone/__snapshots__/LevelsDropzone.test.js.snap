// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LevelsDropzone Check matching snapshot 1`] = `
<div
  className="levelsDropzones"
>
  <Scrollbars
    autoHeight={false}
    autoHeightMax={200}
    autoHeightMin={0}
    autoHide={false}
    autoHideDuration={200}
    autoHideTimeout={1000}
    className="scrollbar"
    hideTracksWhenNotNeeded={false}
    renderThumbHorizontal={[Function]}
    renderThumbVertical={[Function]}
    renderTrackHorizontal={[Function]}
    renderTrackVertical={[Function]}
    renderView={[Function]}
    tagName="div"
    thumbMinSize={30}
    universal={false}
  >
    <div
      className="scrollbarContent"
    >
      <Field
        changeLevelImage={[Function]}
        changeLevelNumber={[Function]}
        component={[Function]}
        key="0"
        name="levels[0]"
        onRemoveLevelClick={[MockFunction]}
        params={
          Object {
            "entityId": undefined,
            "image": undefined,
            "imageLink": null,
            "levelIndex": 0,
            "number": 1,
            "unique": true,
          }
        }
        translate={
          [MockFunction] {
            "calls": Array [
              Array [
                "building.addLevel",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      />
      <Field
        changeLevelImage={[Function]}
        changeLevelNumber={[Function]}
        component={[Function]}
        key="1"
        name="levels[1]"
        onRemoveLevelClick={[MockFunction]}
        params={
          Object {
            "entityId": undefined,
            "image": undefined,
            "imageLink": null,
            "levelIndex": 1,
            "number": 2,
            "unique": true,
          }
        }
        translate={
          [MockFunction] {
            "calls": Array [
              Array [
                "building.addLevel",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      />
      <div />
    </div>
  </Scrollbars>
  <div
    className="addLevelButtonContainer"
  >
    <WithStyles(ForwardRef(Button))
      color="primary"
      onClick={[Function]}
    />
  </div>
</div>
`;
